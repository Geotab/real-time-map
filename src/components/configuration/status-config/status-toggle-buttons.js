import React, { Component } from "react";
import { diagnosticSearch } from "./status-search";

export class StatusToggleButtons extends React.Component {
  constructor(props) {
    super(props);
    this.state = { visibility: true };
    this.toggleVisibility = this.toggleVisibility.bind(this);
  }

  toggleVisibility() {
    this.setState(state => {
      return { visibility: !state.visibility };
    });
  }

  render() {
    return (
      <>
        <button
          id="toggleStatus"
          className={`toggleButton ${
            this.state.visibility ? "shown" : "notShown"
          }`}
          title="Toggle All Statuses"
          onClick={() => {
            this.toggleVisibility();
            if (this.state.visibility) {
              diagnosticSearch.hideAllItems(this.props.setStatusDisplay);
            } else {
              diagnosticSearch.showAllItems(this.props.setStatusDisplay);
            }
          }}
        ></button>
        <button
          id="deleteStatus"
          className="deleteButton"
          title="Delete All Statuses"
          onClick={() =>
            diagnosticSearch.deleteAllItems(this.props.setStatusDisplay)
          }
        ></button>
      </>
    );
  }
}
