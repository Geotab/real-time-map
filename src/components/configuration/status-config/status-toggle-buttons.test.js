import React from "react";
import {StatusToggleButtons} from "./status-toggle-buttons";
import { render, mount, shallow } from "enzyme";

let wrapper;

let state;
beforeEach(() => {
  state = {
    visibility: true 
  };
});

test("should successfully render the component and match screenshot", () => {
  const component = render(<StatusToggleButtons />);
  expect(component).toMatchSnapshot();
});

test("should successfully togglenpm te component state", () => {
  wrapper = shallow(<StatusToggleButtons />);
  const instance = wrapper.instance();
  instance.toggleVisibility();
  expect(wrapper.state.visibility).toBeFalsy();
});