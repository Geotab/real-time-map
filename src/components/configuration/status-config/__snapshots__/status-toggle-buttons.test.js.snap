// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully render the component and match screenshot 1`] = `
Array [
  <button
    class="toggleButton shown"
    id="toggleStatus"
    title="Toggle All Statuses"
  />,
  <button
    class="deleteButton"
    id="deleteStatus"
    title="Delete All Statuses"
  />,
]
`;
